kyaml/runfn/runfn.go

```diff
@@ -70,6 +73,11 @@ func (r RunFns) Execute() error {
 	if err != nil {
 		return err
 	}
+
+	if !r.NoMerge {
+		fltrs = append(fltrs, filters.MergeFilter{})
```

Comment:
the input directory may contain multiple resources that are hard to disambiguate -- e.g. both patches and resources with the name/gvk.

we should update the mergefilter to support using the filepath as part of the merge key (i.e. similar to #2175), and then merge using the filepath after the name has been defaulted

lmk if you want me to add that to mergefilter